lib/util.ts

```diff
@@ -667,6 +667,19 @@ export class ShapeUtil {
     return size;
   }
 
+  /**
+   * Determines the shape of output tensor y = flatten(x, axis)
+   * @param dims - shape of input tensor
+   * @param axis - flatten axis
+   */
+  static flattenShape(dims: ReadonlyArray<number>, axis: number): ReadonlyArray<number> {
+    const total = dims.reduce((x, y) => x * y, 1);
```

Comment:
Do we really need to reduce dims to know the total size ? We actually already have it from the property `inputs[0].length`. I suggest adding an optional parameter `size` in this function that can be total. If unspecified, it can be computed from `dims` using current logic.

Reply:
I prefer keeping the current function's signature as it is consistent with other shapeutils functions in the `Util` package. Computing `total` can be a bit slower, however the logic is easier to understand.